agent/dns.go

```diff
@@ -1896,6 +1900,7 @@ func (d *DNSServer) resolveCNAME(cfg *dnsConfig, name string, maxRecursionLevel
 		resp := &dns.Msg{}
 
 		req.SetQuestion(name, dns.TypeANY)
+		// TODO: handle error response
 		d.dispatch("udp", nil, req, resp, maxRecursionLevel-1)
```

Comment:
This is the last TODO to resolve. Looking at it again, I guess we only ever return the `resp.Answer` here, so ignoring the error might be ok, since the error only impacts the `MsgHdr` and we won't send that anyway.